tests/integ/test_processing.py

```diff
@@ -170,6 +202,90 @@ def test_sklearn_with_customizations(
     assert job_description["StoppingCondition"] == {"MaxRuntimeInSeconds": 3600}
 
 
+def test_sklearn_with_custom_default_bucket(
+    sagemaker_session_with_custom_bucket,
+    image_uri,
+    sklearn_full_version,
+    cpu_instance_type,
+    output_kms_key,
+):
+
+    input_file_path = os.path.join(DATA_DIR, "dummy_input.txt")
+
+    sklearn_processor = SKLearnProcessor(
```

Comment:
IMO: since this functionality lives in the parent `Processor` and propagates down to child classes, the test should use the `Processor` class rather than `SKLearnProcessor`. (If we validate using `Processor`, we validate for all child classes, but if we validate using `SKLearnProcessor`, we don't validate for parent classes.)

Reply:
I see it the other way around.
I prefer to use the implementation in this case (SKLearnProcessor), because it allows us to validate the superclass inherited methods (normalization of inputs) AS WELL as the custom implementation logic (automatic code upload).

Reply:
^ ah, I realize I should've said `ScriptProcessor`, not `Processor` -- if another child class of `ScriptProcessor` is added, and if the integ tests use `ScriptProcessor`, automatic code upload would be tested for that new class by dint of it being tested with `ScriptProcessor`. (For example: what if `SKLearnProcessor` has some code added to it that lets this test pass just for SKLearnProcessor?)